integr8ly/delorean
# add datahub import

Add a `datahub-import` command to import data from a downtime-report to the datahub prometheus instance.

Test data can be seen here http://pushgateway-dh-prod-monitoring.cloud.datahub.psi.redhat.com/ and a grafana dashboard here https://grafana.datahub.redhat.com/dashboard/db/rhmi-product-downtime?orgId=1

To test:
1.Upload a `downtime-report.yaml` to an s3 bucket you have access to ( I can provide access if needed ) 
2.Run this command `./delorean report datahub-import -b <BUCKET-NAME> -p "http://pushgateway-redhat-rhmi-middleware-monitoring-operator.apps.stobin-prom.x0f5.s1.devshift.org/"`
3.See that the data has been pushed here http://pushgateway-redhat-rhmi-middleware-monitoring-operator.apps.stobin-prom.x0f5.s1.devshift.org

This PR includes a bump of prometheus/client_golang to 1.2.0 since that version fixes an upload error in pushgateway 1.10+ which was required.

The datahub grafana deployment does not use the grafana operator so the dashboard must be updated manually going forward and backed up in json form in the `templates/` directory
